etcdserver/corrupt.go

```diff
@@ -24,6 +24,69 @@ import (
 	"github.com/coreos/etcd/pkg/types"
 )
 
+func (s *EtcdServer) checkHashKVInit() {
+	// TODO: separate configuration for initial hash check?
+	if s.Cfg.CorruptCheckTime < 3*time.Second {
+		return
+	}
+
+	select {
+	case <-s.readych: // sync with s.publish
+	case <-s.stopping:
+		return
+	}
+	mbs := s.cluster.Members()
+	n := len(mbs)
+	if n < 3 {
+		plog.Infof("no need to check corruption on %s (%d members)", s.ID().String(), n)
+		return
+	}
+	plog.Infof("corruption checking on %s (%d members)", s.ID().String(), n)
+
+	h, rev, _, err := s.kv.HashByRev(0)
```

Comment:
what happens when this local node is slow, and the revision has been compacted on other peers?